src/lib/component/autocomplete/index.js

```diff
@@ -0,0 +1,141 @@
+import AutocompleteModel from './autocompleteModel'
+import createResultElement from './createResultElement'
+import debounce300 from '../debounce300'
+import delegate from 'delegate'
+
+export default class Autocomplete {
+  #inputElement
+  #onSelect
+  #model
+  #resultsElement
+
+  constructor(inputElement, onSearch, onSelect) {
+    this.#inputElement = inputElement
+    this.#onSelect = onSelect
+
+    this.#model = new AutocompleteModel(
+      (term) => onSearch(term, (results) => (this.#model.items = results)),
+      (items) => this.#renderItem(items),
+      (index) => this.#highlight(index)
+    )
+
+    this.#resultsElement = document.createElement('ul')
+    this.#resultsElement.setAttribute('popover', 'auto')
+    this.#resultsElement.classList.add('textae-editor__dialog__autocomplete')
+    inputElement.parentElement.appendChild(this.#resultsElement)
+
+    const handleInput = debounce300((term) => {
+      this.#model.term = term
+
+      if (term.length < 3) {
+        this.#resultsElement.hidePopover()
+      }
+    })
+
+    this.#inputElement.addEventListener('input', (event) =>
+      handleInput(event.target.value)
+    )
+    this.#inputElement.addEventListener('keydown', (event) =>
+      this.#handleKeydown(event)
+    )
+
+    // Hide popover when input is out of focus.
+    this.#inputElement.addEventListener('blur', () =>
+      this.#resultsElement.hidePopover()
+    )
+
+    this.#setEventHandlerToResults()
+  }
+
+  #setEventHandlerToResults() {
```

Comment:
```suggestion
  #setEventHandlersToResults() {
```

細かいですが、複数のイベントハンドラーを設定するので、複数型がよいと思います。

Reply:
その通りですね。修正しました。
https://github.com/pubannotation/textae/pull/142/commits/1750569697bda8f72bc86c8aa49758c79c41a69f